packages/react-vapor/src/components/select/tests/MultiSelectConnected.spec.tsx

```diff
@@ -1,239 +1,241 @@
+            expect(within(lists[0]).getByText('🥔')).toBeInTheDocument();
+            expect(within(lists[0]).getByText('🍟')).toBeInTheDocument();
 
-            let state = _.findWhere(store.getState().listBoxes, {id});
+            // open the dropdown
+            userEvent.click(screen.getByRole('button', {name: 'Select an option'}));
 
-            expect(state.selected).toEqual([firstSelected, secondSelected]);
+            lists = screen.getAllByRole('list');
+            expect(within(lists[1]).getByText('🌱')).toBeInTheDocument();
+            expect(within(lists[1]).queryByText('🥔')).not.toBeInTheDocument();
+            expect(within(lists[1]).queryByText('🍟')).not.toBeInTheDocument();
+        });
 
-            multiSelect.find(SelectedOption).at(0).find('.remove-option').simulate('click');
+        it('is does not open the dropdown if there is no unselected items', () => {
+            const items = [
+                {value: '🌱', selected: true},
+                {value: '🥔', selected: true},
+            ];
+            render(<MultiSelectConnected id={id} items={items} />);
 
-            state = _.findWhere(store.getState().listBoxes, {id});
+            // open the dropdown
+            userEvent.click(screen.getByRole('button', {name: 'Select an option'}));
 
-            expect(state.selected).toEqual([secondSelected]);
+            // one list for the selected values
+            expect(screen.getAllByRole('list').length).toBe(1);
         });
 
-        it('should remove all selected items when the clear button is clicked', () => {
-            const firstSelected = 'dis 1';
-            const secondSelected = 'dis two';
+        it('is does not open the dropdown if there is no items', () => {
+            render(<MultiSelectConnected id={id} items={[]} />);
 
-            mountMultiSelect([
-                {value: 'a'},
-                {value: firstSelected, selected: true},
-                {value: secondSelected, selected: true},
-            ]);
+            // open the dropdown
+            userEvent.click(screen.getByRole('button', {name: 'Select an option'}));
 
-            let state = _.findWhere(store.getState().listBoxes, {id});
+            // one list for the selected values
+            expect(screen.getAllByRole('list').length).toBe(1);
+        });
 
-            expect(state.selected).toEqual([firstSelected, secondSelected]);
+        it('allows custom toggleClasses on the dropdown', () => {
+            const {container} = render(<MultiSelectConnected id={id} items={[]} toggleClasses="coulili-zazou" />);
 
-            multiSelect.find('.remove-all-selected-options').simulate('click');
+            expect(container.querySelector('.coulili-zazou')).toBeInTheDocument();
+        });
 
-            state = _.findWhere(store.getState().listBoxes, {id});
+        describe('Sortable', () => {
+            const dragAndDrop = (source: Element, target: Element) => {

```

Comment:
Is this method reusable for testing other components? If so, should we put it somewhere more common?

Reply:
It could be reused elsewhere but I didn't know where to put it and if it was really useful. We have drag and drop in only two components